pages/index.js

```diff
@@ -1,68 +1,67 @@
-import envFlag from 'node-env-flag';
 import React from 'react';
-import Meta from '../frontend/components/meta';
-import Header from '../frontend/components/header';
-import SuggestionAndSearch from '../frontend/components/suggestion-and-search';
+import { HashRouter, StaticRouter, Route, Link } from "react-router-dom";
 import { BadgeExamples } from '../frontend/components/badge-examples';
-import MarkupModal from '../frontend/components/markup-modal';
-import Usage from '../frontend/components/usage';
-import Footer from '../frontend/components/footer';
-import badgeExampleData from '../badge-examples.json';
-import { prepareExamples, predicateFromQuery } from '../frontend/lib/prepare-examples';
+import ExamplesPage from '../frontend/components/examples-page';
+import { baseUri, longCache } from '../frontend/constants';
 
-const baseUri = process.env.BASE_URL;
-const longCache = envFlag(process.env.LONG_CACHE, false);
-
-export default class IndexPage extends React.Component {
-  state = {
-    query: null,
-    example: null
-  };
+class IndexPage extends ExamplesPage {
```

Comment:
React is based on a one-way flow, where props and state are used to generate the component tree in `render()`. In particular, invoking `render()` should not result in a change to the state.

Inheritance between components is not a common pattern. Functions which generate components (called "higher-order components") are more common, though they probably are not needed here. I think you can accomplish the code and logic by factoring out components and helper functions, and using composition.

Here are some thoughts:

- You could create SharedHeader and SharedFooter components with the bits that are shared between the two pages. That way you can move the code from `getBody()` back into `render()` without duplicating anything.
- The SuggestionAndSearch component has a `queryChanged` prop. Passing your own function into that component is a clearer way for that component to emit its state changes back up the tree. You can pass that prop through SharedHeader if need be. In SuggestionAndSearch might be necessary to recompute `queryChangedDebounced`, perhaps in `componentWillReceiveProps`. Otherwise the notifications will keep being sent to the old function.
- Consider whether it might be possible to move some of the (shared) timeout logic into SuggestionAndSearch.

This is what I'm coming up with by _thinking_ about this problem, not by trying to make the changes… so take 'em with a hefty sprinkle of salt!